traits.go

```diff
@@ -0,0 +1,92 @@
+package analytics
+
+import "time"
+
+// This type is used to represent traits in messages that support it.
+// It is a free-form object so the application can set any value it sees fit but
+// a few helper method are defined to make it easier to instantiate traits with
+// common fields.
+// Here's a quick example of how this type is meant to be used:
+//
+//	analytics.Identify{
+//		UserId: "0123456789",
+//		Traits: analytics.NewTraits()
+//			.SetFirstName("Luke")
+//			.SetLastName("Skywalker")
+//			.Set("Role", "Jedi"),
+//	}
+//
+type Traits map[string]interface{}
+
+func NewTraits() Traits {
+	return make(Traits, 10)
+}
+
+func (t Traits) SetSetAddress(address string) Traits {
+	return t.Set("address", address)
+}
+
+func (t Traits) SetAge(age int) Traits {
+	return t.Set("age", age)
+}
+
+func (t Traits) SetAvatar(url string) Traits {
+	return t.Set("avatar", url)
+}
+
+func (t Traits) SetBirthday(date time.Time) Traits {
+	return t.Set("birthday", date)
+}
+
+func (t Traits) SetCreatedAt(date time.Time) Traits {
+	return t.Set("createdAt", date)
+}
+
+func (t Traits) SetDescription(desc string) Traits {
+	return t.Set("description", desc)
+}
+
+func (t Traits) SetEmail(email string) Traits {
+	return t.Set("email", email)
+}
+
+func (t Traits) SetFirstName(firstName string) Traits {
+	return t.Set("firstName", firstName)
+}
+
+func (t Traits) SetGender(gender string) Traits {
+	return t.Set("gender", gender)
+}
+
+func (t Traits) SetID(id string) Traits {
```

Comment:
I think this is deprecated and should be removed (people should send userid instead).